### PR TITLE
feature: add category choice to quiz

### LINKED ISSUE #657: [Feature] - Add ability for users to choose a category
### What feature would you like to see?

We now have enough questions to break up the quiz into categories. When the user clicks on the button to start the quiz, they should be directed to this screen here.

## First Slide
Screen Shot 2023-01-12 at 8 51 57 AM

Then once a user selects a category, they will be directed to this screen here.

## Second Slide
Screen Shot 2023-01-12 at 8 47 44 AM

Then once a user selects a question amount, they will be directed to the actual quiz here. 

## Quiz
Screen Shot 2023-01-12 at 8 48 10 AM

If you are interested in working on this issue, then please reply back to this issue and it will be assigned to you.

### PR DESCRIPTION
**Pull request by BekahHW**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

Closes #657 

This PR adds the SelectCategory component, which allows users to choose a category for the quiz. It also moves the array of categories to `QuizTemplate` for better organization. After selecting a category, the quiz number is displayed. 

The conditional rendering of questions has been reorganized to ensure they are only rendered after selecting a number. Constants have been reworked for improved readability. The filtered category is now used to start the quiz. 

Tests have been added to ensure the component renders and displays options.

_Generated using [OpenSauced](https://opensauced.ai/)._

### CODE REVIEW

**Comment by jdwilkin4** [comment]
Hey @BekahHW ! 

Thanks for all of your hard work on this.
I will have time to review it after work 👍

**Comment by BekahHW** [comment]
Thanks, @jdwilkin4. Take your time. Just wanted to make sure you saw it.

File: src/__tests__/SelectCategory.test.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "JavaScript",
+      "Accessibility",
+      "General CS",
+      "IT",
+      "Linux",
+      "Python",
+      "SQL"
+    ];
+  
+    it("renders without crashing", () => {
```
Comment (jdwilkin4) [inline src/__tests__/SelectCategory.test.tsx:20]: We should change the title of this test to sometime else like "displays the Choose a Category screen"

File: src/components/SelectQuiz.tsx [hunk 2]
```diff
@@ -3,25 +3,33 @@ import React from "react";
...
+    value={choice}
+    key={index}
+  >
+    {choice}
+  </button>
+))}
+           
+            <button className="select-btns" onClick={SelectQuizProps.startQuiz} value={SelectQuizProps.totalQuestions}>
+              All ({SelectQuizProps.totalQuestions})
             </button>
```
Comment (jdwilkin4) [inline src/components/SelectQuiz.tsx:31]: It looks like the formatting is a little off here
I would suggest running `npm run format`

File: src/components/QuizTemplate.tsx [hunk 3]
```diff
@@ -75,10 +113,13 @@ const QuizTemplate: React.FC<QuizProps> = QuizProps => {
   };
 
   const resetQuiz = () => {
-    setQuiz(ALL_CATEGORIES);
+    setSelectedCategory(""); // Reset selected category
+  setSelectedQuiz(0); // Reset selected quiz
+    // setQuiz(ALL_CATEGORIES);
```
Comment (jdwilkin4) [inline src/components/QuizTemplate.tsx:118]: we could get rid of this line since it is commented out

File: src/constants.ts [hunk 4]
```diff
@@ -15,25 +15,29 @@ import qualityAssuranceQuiz from "./data/quality-assurance-quiz";
...
-  ...sqlQuiz,
-  ...agileQuiz,
-  ...qualityAssuranceQuiz,
-  ...securityQuiz,
-  ...regexQuiz
+  ...mapQuizCategory(accessibilityQuiz, "Accessibility"),
+  ...mapQuizCategory(cssQuiz, "CSS"),
+  ...mapQuizCategory(cloudComputingQuiz, "Cloud Computing"),
+  ...mapQuizCategory(devopsQuiz, "DevOps"),
+  ...mapQuizCategory(freecodecampQuiz, "FreeCodeCamp"),
```
Comment (jdwilkin4) [inline src/constants.ts:26]: misspelling of freeCodeCamp

```suggestion
  ...mapQuizCategory(freecodecampQuiz, "freeCodeCamp"),
```

**Review by jdwilkin4 — CHANGES_REQUESTED** [review]
I tested it locally with a few quizzes and it looks good.
I left a few comments. 👍 

But also, some of the tests for the `SelectQuiz.test.tsx` are not passing on my end.

**Review by jdwilkin4 — APPROVED** [review]
ready to ship 🚢 
good work on this feature 😃

**Pull request merged**